### DISCUSSION TITLE
Window often doesn't update after first frame unless the window is resized

### DISCUSSION THREAD

**Discussion by jcarvajal288** [bevyengine/bevy#14810] [Q&A]
A recurring problem I'm having is that when I run the game I'm making, it often (but not always) only shows the first frame.  The game progresses while it's frozen like this, but it doesn't show any update until I resize the window, after which it catches up to where the game should be at that point.

I'm unsure what part of my code is responsible for this, but it happened more frequently after I added another Update system on top of my FixedUpdate system.  Here's my main function if it helps:
```
fn main() {
    App::new()
        .add_plugins(DefaultPlugins.set(WindowPlugin {
            primary_window: Some(Window {
                resolution: WindowResolution::new(736., 736.).with_scale_factor_override(1.0),
                ..default()
            }),
            ..default()
        }))
        .insert_resource(Time::<Fixed>::from_duration(Duration::from_millis(500)))
        .insert_resource(Images::default())
        .insert_resource(level_map::load_level_1())
        .add_systems(Startup, (load_images, setup).chain())
        .add_systems(FixedUpdate, snake::snake_movement_system)
        .add_systems(Update, snake::change_direction_system)
        .run();
}
fn setup(mut commands: Commands, level_map: Res<LevelMap>, images: Res<Images>, windows: Query<&Window>) {
    commands.spawn(Camera2dBundle::default());

    let window_center = Vec2::new(windows.single().resolution.width() / 2., windows.single().resolution.height() / 2.);

    snake::spawn_snake(&mut commands, &images, window_center);

    level_map.draw(commands, &images, window_center);
}
```

And here's a link to the full code: https://github.com/jcarvajal288/snake

Any idea what the problem could be?

**Comment by jcarvajal288** [comment]
It seems to have something to do with Wayland.  I haven't seen the problem yet on X11.

**Comment by PraxTube** [comment]
I am having the same issue (am also on wayland, though not sure if that is the reason). It only happens on my laptop though, on my PC it works fine everytime (I have arch on both with pretty much identical configs). Only difference is my laptop is using NVIDEA GPU and intel CPU (also laptop is just worse in terms of performance).

@jcarvajal288  did you figure out why this is happening?

**Reply by christopher-r-anderson** [reply]
@PraxTube @jcarvajal288 I had this problem as well. It looks like you can do one of two things:

1.  Enable the wayland feature so that things are not going through X11: https://bevy-cheatbook.github.io/platforms/linux.html#x11-and-wayland
2. Wait for 0.15 where I think this has been addressed here: https://github.com/bevyengine/bevy/pull/16123

**Reply by PraxTube** [reply]
Hey man, that did the trick, thanks! I really appreciate you sharing this <3
